distributed/protocol/cupy.py

```diff
@@ -85,15 +87,35 @@ def dask_deserialize_cupy_ndarray(header, frames):
 try:
     from cupy.cusparse import MatDescriptor
     from cupyx.scipy.sparse import spmatrix
-
-    cupy_sparse_types = [MatDescriptor, spmatrix]
 except ImportError:
-    cupy_sparse_types = []
+    MatDescriptor = None
+    spmatrix = None
+
+
+if MatDescriptor is not None:
+
+    def reduce_matdescriptor(other):
+        # Pickling MatDescriptor errors
+        # xref: https://github.com/cupy/cupy/issues/3061
+        return cupy.cusparse.MatDescriptor.create, ()
+
+    copyreg.pickle(MatDescriptor, reduce_matdescriptor)
```

Comment:
`MatDescriptor` goes through normal pickling. However that will cause a segfault (as it will contain an invalid `int` that points to invalid memory). So we special case pickling to avoid this.

xref: https://github.com/cupy/cupy/issues/3061